services/imap/imap.go

```diff
@@ -0,0 +1,480 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package imap
+
+import (
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+	"mime"
+	net_mail "net/mail"
+	"strings"
+	"sync"
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
+
+	"github.com/emersion/go-imap"
+	"github.com/emersion/go-imap/client"
+	"github.com/emersion/go-message"
+	"github.com/emersion/go-message/charset"
+	"github.com/emersion/go-message/mail"
+	"golang.org/x/text/encoding/simplifiedchinese"
+)
+
+func init() {
+	charset.RegisterEncoding("gb18030", simplifiedchinese.GB18030)
```

Comment:
Because this special format has been used in some mail server but not been addded by default. maybe need add more.  In fact all other charset will be convert to ``uft-8``, so not need add it.